text/0911-const-fn.md

```diff
@@ -172,6 +172,18 @@ const fn arithmetic_ops<T: Int>() -> [fn(T, T) -> T; 4] {
 }
 ```
 
+`const` functions can also be unsafe, allowing construction of types that require
+invariants to be maintained (e.g. `std::ptr::Unique` requires a non-null pointer)
+```rust
+struct OptionalInt(u32);
+impl OptionalInt {
+    /// Value must be non-zero
+    unsafe const fn new(val: u32) -> OptionalInt {
```

Comment:
I agree with @petrochenkov: `unsafe fn` feels like a more fundamental concept/thing than `const fn`. (`const` on a function doesn't outlaw it from being used as a non-`const` function, while `unsafe` does outlaw it from being used as a non-`unsafe` one.)